actors/builtin/miner/miner_actor.go

```diff
@@ -857,6 +861,59 @@ func (a Actor) DeclareFaultsRecovered(rt Runtime, params *DeclareFaultsRecovered
 	return nil
 }
 
+func (a Actor) processSkippedFaults(rt Runtime, st State, skipped *bitfield.BitField, deadline *DeadlineInfo,
+	epochReward abi.TokenAmount, qaPowerTotal abi.StoragePower) ([]*SectorOnChainInfo, abi.TokenAmount) {
+
+	empty, err := skipped.IsEmpty()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "failed to check if skipped sectors is empty")
+	if empty {
+		return nil, abi.NewTokenAmount(0)
+	}
+
+	currEpoch := rt.CurrEpoch()
+	store := adt.AsStore(rt)
+	var skippedFaultSectors []*SectorOnChainInfo
+
+	deadlines, err := st.LoadDeadlines(adt.AsStore(rt))
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deadlines")
+
+	// Check that the declared sectors are actually due at the deadline.
+	deadlineSectors := deadlines.Due[deadline.Index]
+	contains, err := abi.BitFieldContainsAll(deadlineSectors, skipped)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "could not check if skipped faults are in deadline")
+	if !contains {
+		rt.Abortf(exitcode.ErrIllegalArgument, "skipped faults contains sectors not due in deadline %d", deadline.Index)
+	}
+
+	// Find all skipped faults that have been labeled recovered
+	retractedRecoveries, err := bitfield.IntersectBitField(st.Recoveries, skipped)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to intersect sectors with recoveries")
+
+	// Ignore skipped faults that are already faults but have not been declared recovered
+	newFaults, err := bitfield.SubtractBitField(skipped, st.Faults)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to subtract existing faults from skipped")
+	penalizableFaults, err := bitfield.MergeBitFields(newFaults, retractedRecoveries)
```

Comment:
Blocking: As noted in the issue, I think we'll need to charge penalty tokens for retracted recoveries, but not deduct power, so we'll need to track these carefully separately. 

The return value `skippedFaultSectors` currently contains all of these. We need to load sector info for retractions to calculate fee, but exclude from power.